drivers/gpio/gpio_intel_apl.c

```diff
@@ -430,40 +430,41 @@ static const struct gpio_intel_apl_config gpio_intel_apl_cfg = {
 	.islands = {
 		{
 			/* North island */
-			.reg_base = INTEL_APL_GPIO_0_BASE_ADDRESS_0,
+			.reg_base = INTEL_APL_GPIO_D0C50000_BASE_ADDRESS_0,
```

Comment:
There should be a dts_fixup.h file that maps what is generated to something this driver would use.   Should be something like CONFIG_APL_GPIO_x_BASE_ADDR (where '_x_' would be something if there is more than one port or block of GPIOs).

Reply:
Added these into the SoC's existing dts_fixup.h